### PR TITLE
Welcome page tweaks

### PR DESCRIPTION
**Pull request by ryanml**
This includes a view small design tweaks targeted for mobile per design review:

1. Reduces the font size of the hero header/subtitle on smaller devices
2. On touchscreen devices removes the outer settings page wrapper/desktop header
3. Reduces hero padding for smaller devices

### CODE REVIEW

File: src/components/buttonsIndicators/button/index.tsx [hunk 1]
```diff
@@ -21,6 +21,8 @@ export interface Props {
   onClick?: () => void
   id?: string
   disabled?: boolean
+  color?: string
+  heavy?: boolean
```
Comment (ryanml) [inline src/components/buttonsIndicators/button/index.tsx]: These additions were requested by @jenn-rhim 

These are simple, non-breaking overrides that let you specify a certain color for a button regardless of its Type, and to explicitly use `font-weight: 700` for an extra bold texted button

Comment (NejcZdovc) [reply src/components/buttonsIndicators/button/index.tsx]: cc @petemill because this is completely against what we defined in language and on our meetings

Comment (ryanml) [reply src/components/buttonsIndicators/button/index.tsx]: I think this warrants a discussion for the next front-end component meeting:

We have this design goal for the Welcome Page for example, and this includes certain weights and color changes that violate what is in our standard button component. I am in between a rock and a hard place trying to both respect the shared components and implement the product as delivered. It will be good to be on the same page with this.

Comment (NejcZdovc) [reply src/components/buttonsIndicators/button/index.tsx]: totally agree, design and development need to be in sync with this one, on the same page. that means everybody

Comment (ryanml) [reply src/components/buttonsIndicators/button/index.tsx]: In the mean time I am going to revert the Button related portions of this PR, and we can review based on that. A follow up PR will be produced if necessary as the result of the above meeting.

Comment (jenn-rhim) [reply src/components/buttonsIndicators/button/index.tsx]: We need to add 'white' buttons, both solid and outlined, in the library for the cases where the buttons are placed on the (dark) colored background. We already have many cases in production using this style that we can benefit.

Comment (NejcZdovc) [reply src/components/buttonsIndicators/button/index.tsx]: then we need to just add it to design language and then add it to shared component as predefined type. As soon as we will add option for custom color this will be abused IMHO

Comment (petemill) [reply src/components/buttonsIndicators/button/index.tsx]: Hey @jenn-rhim @ryanml - good news: we have a proposed plan for this that hopefully solves all the requirements, and probably needs re-presenting. One of the reasons we don't want to just throw around arbitrary colors (aside from maintaining consistency and change-ability), is still supporting 'themes' e.g. for dark mode. Bad news: the plan has not yet been implemented because we're yet to be able to get buy-in from everyone and adjust the process to use theme properties properly. I'm not sure I'll do the plan justice in this comment, but I'll attempt it:

Right now we have Button variations that are defined by semantic names, that result in different colors: 
image

The problem described here is that sometimes these buttons are on dark background and sometimes are on light background. Obviously the Default Button (which currently results in a Black button) won't look good on a dark background. The design requirement is that dark backgrounds would want different color Buttons for the same states. Right now, these darker backgrounds are used in specific known places. However, now / soon, the user will be able to make _some_ currently-light backgrounds in to dark backgrounds by changing theme preference.

So the plan is to have a concept of _theme containers_, that will know whether they are dark or light, some are always dark or light, and some will be changed as a result of user theme choice (dark or light). Essentially each part of a layout would be one of 3 things as defined by a designer:
- Always Light
- Always Dark
- Light / Dark depending on preference
(and of course the container itself would be able to have arbitrary and custom background color / graphic / whatever)

In these containers, the theme will use either Brave-Light values or Brave-Dark values depending on which theme the container is set to.

Therefore a 'Default' Button will be Black in a light container and White in a dark container. The Button would still be defined semanticly and wouldn't be able to be 'any' color over the variations that the team has defined:

| Colors        | Light-mode value           | Dark-mode value  |
| ------------- |:-------------:| -----:|
|    brandBraveInteracting | colors.orange500  |  ??? |
|    brandBraveActive | colors.orange200  |  ??? |
|    brandBraveLight | colors.orange000  |  ??? |
|     |
|    brandBatInteracting | colors.blurple500  |  ??? |
|    brandBatActive | colors.blurple200  |  ??? |
|     |
|    defaultControl | colors.grey800  | ??? |
|    defaultControlInteracting | colors.black  |  ??? |
|    defaultControlActive | colors.grey700  |  ??? |
|     |
|    warn | colors.red500  |  ??? |
|    warnInteracting | colors.red600  |  ??? |
|    warnActive | colors.red700  |  ??? |
|     |
|    subtle | colors.grey400  |  ??? |
|    subtleInteracting | colors.grey500  |  ??? |
|    subtleActive | colors.grey600  |  ??? |
    

This probably needs more of a discussion, and if this is an urgent requirement, we can hard-code a color for now, but it'd be great to ratify a solution similar to the one I've described here soon. Especially since we should be quite protective of the Button as it's an example of the rules we want to implement in other controls that allow us to reach the goals of the design system.

File: src/components/buttonsIndicators/button/style.ts [hunk 2]
```diff
@@ -116,7 +117,7 @@ export const StyledText = styled<Props, 'div'>('div')`
   align-items: center;
   text-align: center;
   letter-spacing: 0;
-  font-weight: 500;
+  font-weight: ${p => p.heavy ? '700' : '500'};
```
Comment (petemill) [inline src/components/buttonsIndicators/button/style.ts]: @jenn-rhim @karenkliu what semantically makes a Button get heavy text? Should all Primary action buttons be heavy for example? If it's more arbitrary than that, can we get a 1-sentence rule describing the reason it's used for this 1 instance here and perhaps something semantic will come out of that?

**Comment by ryanml** [comment]
@petemill @NejcZdovc @jenn-rhim there is some great discussion on here pertaining to the button issue (one that will be fully resolved in tomorrows meeting I hope), but the important thing in this PR is some of the spacing updates to the welcome page.

I'm going to revert the button changes from this PR, and then once we are all on the same page, we can do Button component updates in its own PR with own issue. For now we can just look at the other welcome page updates.

cc: @anthonypkeane @gdregalo

File: src/features/rewards/welcomePage/index.tsx [hunk 3]
```diff
@@ -84,9 +84,16 @@ class WelcomePage extends React.PureComponent<Props, {}> {
     this.props.optInAction()
   }
 
+  isTouchscreen = () => {
+    return 'ontouchstart' in document.documentElement
```
Comment (petemill) [inline src/features/rewards/welcomePage/index.tsx]: this doesn't change, so can be a const at the module level, save re-evaluating (admittedly very very quick re-evaluation!)

File: src/features/rewards/welcomePage/style.ts [hunk 4]
```diff
@@ -71,13 +76,21 @@ export const StyledBackground = styled<{}, 'div'>('div')`
...
 `
 
 export const StyledRewardsTitle = styled(Heading)`
   font-weight: 500;
   color: #FFF;
   display: inline-block;
   margin: 17px 0 4px;
+
+  @media (max-width: 360px) {
+    font-size: 36px;
```
Comment (petemill) [inline src/features/rewards/welcomePage/style.ts]: Should this responsive behavior be built in to Heading?

File: src/features/rewards/welcomePage/style.ts [hunk 5]
```diff
@@ -102,6 +115,10 @@ export const StyledSubTitle = styled(Heading)`
   text-align: center;
   line-height: 28px;
   margin: 18px 0 7px;
+
+  @media (max-width: 360px) {
+    font-size: 22px;
```
Comment (petemill) [inline src/features/rewards/welcomePage/style.ts]: Same re: Heading. Otherwise is there much point in using the Heading component if we're setting the font-size here?

File: src/features/rewards/welcomePage/index.tsx (hunk 3 above)
Comment (ryanml) [reply src/features/rewards/welcomePage/index.tsx]: Done

**Comment by ryanml** [comment]
@petemill now using custom StyledComponents for the headers in question, as adding media queries at the Heading module level may have unintended results elsewhere

File: src/features/rewards/hero/index.tsx [hunk 6]
```diff
@@ -7,15 +7,19 @@ import { StyledHero } from './style'
 
 export interface Props {
   id?: string
+  roundBorder?: boolean
```
Comment (NejcZdovc) [inline src/features/rewards/hero/index.tsx]: I think that we should be using isMobile or something like this flag, so that we can use it for other changes as well if needed

File: src/features/rewards/welcomePage/style.ts [hunk 7]
```diff
@@ -71,13 +77,24 @@ export const StyledBackground = styled<{}, 'div'>('div')`
...
   margin: 5px auto 0;
   height: 152px;
+
+  @media (max-width: 460px) {
+    height: 100px;
+  }
 `
 
-export const StyledRewardsTitle = styled(Heading)`
+export const StyledRewardsTitle = styled<{}, 'span'>('span')`
```
Comment (NejcZdovc) [inline src/features/rewards/welcomePage/style.ts]: Why was this changed from Heading?

File: src/features/rewards/welcomePage/style.ts [hunk 8]
```diff
@@ -96,12 +113,19 @@ export const StyledCenterTitle = styled(Heading)`
   margin: 0;
 `
 
-export const StyledSubTitle = styled(Heading)`
+export const StyledSubTitle = styled<{}, 'span'>('span')`
```
Comment (NejcZdovc) [inline src/features/rewards/welcomePage/style.ts]: Same as above

File: src/features/rewards/welcomePage/style.ts (hunk 7 above)
Comment (ryanml) [reply src/features/rewards/welcomePage/style.ts]: I had discussed this with @petemill 

The font size for a couple of the items in the welcome header needs to change based on screen size. Adding this media query override kind of defeats the purpose of using the `Heading` component (which shouldn't be changed). Adding this change to the `Heading` component style could have potentially unrelated consequences, so this seemed the easiest thing to do.

File: src/features/rewards/hero/index.tsx (hunk 6 above)
Comment (ryanml) [reply src/features/rewards/hero/index.tsx]: Sure, can change

File: src/features/rewards/welcomePage/style.ts (hunk 7 above)
Comment (NejcZdovc) [reply src/features/rewards/welcomePage/style.ts]: you can still use header component like was before and just add media query. This way you use standard component and have custom mobile
